src/frontend/js/pages/DashboardCourses/useOrdersEnrollments.tsx

```diff
@@ -0,0 +1,277 @@
+import { useEffect, useRef, useState } from 'react';
+import { QueryKey, useQueryClient } from '@tanstack/react-query';
+import { defineMessages, useIntl } from 'react-intl';
+import { useJoanieApi } from 'contexts/JoanieApiContext';
+import { Resource, ResourcesQuery } from 'hooks/useResources';
+import { Enrollment, Order, PaginatedResponse } from 'types/Joanie';
+import { Maybe } from 'types/utils';
+
+export enum DataType {
+  ENROLLMENT = 'enrollment',
+  ORDER = 'order',
+}
+
+export type Data = {
+  type: DataType;
+  item: Enrollment | Order;
+};
+
+const CHUNK_SIZE = 50;
+
+const messages = defineMessages({
+  errorGet: {
+    id: 'hooks.useOrdersEnrollments.errorGet',
+    description: 'Error message shown to the user when orders or enrollments fetch request fails.',
+    defaultMessage: 'An error occurred while fetching orders and enrollments. Please retry later.',
+  },
+});
+
+const checkIntegrity = (previousStack: Data[], newStack: Data[], integrityIndex: number) => {
+  if (previousStack.length < integrityIndex || newStack.length < integrityIndex) {
+    return false;
+  }
+
+  // If we find a difference, we return false.
+  return !newStack.slice(0, integrityIndex).find((item, index) => {
+    return previousStack[index].item.id !== item.item.id;
```

Comment:
1. That's maybe caused to the ` - 1` included in the previous suggestion. Anyway, let's talk about this in more details during the daily meeting. 

2. Totally! I didn't provided it because it's the default order, but you're right, I"ll add it :)